### ISSUE TITLE
Use of the API to export and import application contexts

### ISSUE THREAD

**Issue by catmorales** [georchestra/mapstore2-georchestra#368]
Could you explain the most relevant method to export an application context (with all its extensions enabled) from one domain to another?
We can easily migrate the maps (in json) but here it is the context with its extensions and their settings.

We assume that the installations are identical in both domains with the same geoserver layers.

Example: 
https://portail-test.sig.rennesmetropole.fr/mapstore/#/context/master-voirie

to 

https://portail.sig.rennesmetropole.fr/mapstore/#/context/master-voirie

**Comment by landryb** [comment]
i think contexts are also json, but they might reference other resources (images, maps) so exporting a context might be several files... i second this need to 'migrate' contexts from envs to envs (dev/stage/prod/etc..)

**Comment by catmorales** [comment]
Context are stored in database in our case.

**Comment by landryb** [comment]
yes, they are stored in the database as json, and the REST API allows to access that json, but iirc a context also references other resources.

**Comment by landryb** [comment]
after looking a bit, a context is scattered in several places / objects in the db (several rest calls done when loading the context, and i might have missed some):
- json for the parent context itself/plugins config in `gs_stored_data` table
- json for the map inside the context in `gs_stored_data` table
- json for attributes/map details config (`gs_attribute` table)
- HTML for map details in `gs_stored_data` table
- ACLs on the context in`gs_security` table
- title/descr/category in `gs_resource` table

so if you wanted to 'migrate' a context from an instance to another, there are several things to copy, and the SQL sequence giving the ids might also get confused.

**Comment by MaelREBOUX** [comment]
Clearly a bad idea to play directly with the database...

**Comment by landryb** [comment]
oh, and i finally found out how a map object is linked to a context..
```
georchestra=> select * from mapstore.gs_attribute where name='context';
 id  | attribute_date |  name   | attribute_number | attribute_text | attribute_type | resource_id
-----+----------------+---------+------------------+----------------+----------------+-------------
  83 |                | context |                  | 75             | STRING         |          78
  90 |                | context |                  | 75             | STRING         |          85
 110 |                | context |                  | 75             | STRING         |         105
 129 |                | context |                  | 75             | STRING         |         117
```
`resource_id` is the id of an entry where `category=MAP` in `gs_resource`, and 75 here is the `resource_id` of the context entry (with `category=CONTEXT`) - the json itself being stored in `gs_stored_data` table.

i also realized you can view a 'context child map' (eg `/context/$context_name/$resource_id`) outside of its context, by simply opening `/viewer/openlayers/$resource_id` the map will open with the default context tools.
